### DISCUSSION TITLE
Mutation Tracking doesn't trigger append event

### DISCUSSION THREAD

**Discussion by adam-mrozik** [sqlalchemy/sqlalchemy#8971] [Usage Questions]
Hello All,

I have this kind of field in my model
```
some_field = Column(MutableList.as_mutable(ARRAY(Integer)), default=[], index=True)
```

and I have an event for it:
```
@event.listens_for(User.some_field, 'append')
```

Unfortunately, this event does never get triggered. After some thorough testing, the only attribute event I was able to trigger upon append for this field was `modified` event. The problem with this though is that it doesn't return appended value (or old vs new like in `set`), so I am unable to trigger an action specifically on the appended item.

Not sure if this is a bug or me missing something. I suspect that because `MutableList` only records `self.changed()` for every action, it doesn't differentiate between append, remove or anything else.

I would really appreciate help on this issue.

Thanks!

**Comment by zzzeek** [comment]
the "append" event only applies to collections mapped with relationship().  If you want to track things being appended to the list you have on a single mapped column, you have to subclass the list or MutableList class itself, there is no "event" hook within sqlalchemy.ext.mutable for this kind of thing (there maybe could be someday, but it's not a thing right now).
